src/openassetio-python/tests/bridge/python/test_converter.cpp

```diff
@@ -80,25 +83,26 @@ SCENARIO("Mutations in one language are reflected in the other") {
         Py_DECREF(resultBool);
       }
     }
-    WHEN("Data is set via the python object") {
+
+    WHEN("data is set via the Python object") {
       PyObject_CallMethod(pyTraitsData, "addTrait", "s", kTestTraitId);
-      CHECK(traitsData->hasTrait("") == false);
+      THEN("C++ object reflects the data set") { CHECK(traitsData->hasTrait(kTestTraitId)); }
     }
     Py_DECREF(pyTraitsData);
   }
 }
 
 SCENARIO("Casting to PyObject extends object lifetime") {
-  GIVEN("A python object casted from a C++ object") {
-    WHEN("The C++ object falls out of scope") {
-      PyObject* pyTraitsData = nullptr;
-      {
-        TraitsDataPtr traitsData = TraitsData::make();
-        traitsData->addTrait(kTestTraitId);
-        pyTraitsData = openassetio::python::converter::castToPyObject(traitsData);
-        CHECK(Py_REFCNT(pyTraitsData) == 1);
-      }
-      THEN("The python object remains alive and can be operated on via the python interpreter") {
+  GIVEN("A Python object casted from a C++ object") {
+    TraitsDataPtr traitsData = TraitsData::make();
+    traitsData->addTrait(kTestTraitId);
+    PyObject* pyTraitsData = openassetio::python::converter::castToPyObject(traitsData);
+
+    WHEN("C++ reference is destroyed") {
+      CHECK(Py_REFCNT(pyTraitsData) == 1);  // Initial condition.
+      traitsData.reset();
```

Comment:
:+1: Much better, wasn't possible when we were using Manager due to there not being a default constructor